event_types/event_types.py

```diff
@@ -584,79 +584,104 @@ def bin_data_in_energy(dtf, n_bins=20, log_e_reco_bins=None, return_bins=False):
+        elif offset_bins != offset_bins.sort():
+            raise ValueError("Offset bins must be in increasing order.")
 
 
 def bin_data_in_energy_and_offset(
-    dtf, n_bins=20, log_e_reco_bins=None, offset_bins=None, return_bins=False
+        dtf, n_e_bins=20, n_offset_bins=1, log_e_reco_bins=None, offset_bins=None, return_bins=False
 ):
     """
-    Bin the data in dtf to n_bins with equal statistics.
+    Bin the data in dtf into log energy and camera offset bins.
 
     Parameters
     ----------
     dtf: pandas DataFrame
         The DataFrame containing the data.
         Must contain a 'log_reco_energy' column (used to calculate the bins).
-    n_bins: int, default=20
+    n_e_bins: int, default=20
         The number of reconstructed energy bins to divide the data in.
-    log_e_reco_bins: array-like, None
-        In case it is not none, it will be used as the energy bins to divide the data sample
-    offset_bins: array-like
-        Camera-offset bins (in degrees) to divide the data sample.
-
+    n_offset_bins: int, default=1
+        The number of reconstructed camera-offset bins to divide the data in.
+    log_e_reco_bins: array-like, default=None
+        In case it is not None, it will be used as the log energy bins to divide the data sample
+    offset_bins: array-like, default=None
+        In case it is not None, it will be used as the camera-offset bins (in degrees) to divide
+        the data sample.
     return_bins: bool
-        If true, the function will return the log_e_reco_bins used to bin the data.
+        If true, the function will return the log_e_reco_bins and offset_bins used to bin the data.
 
     Returns
     -------
-    A dictionary of DataFrames (keys=energy ranges, values=separated DataFrames).
+    dtf_e_offset: nested dict of DataFrames
+        1st dict:
+            keys=energy ranges, values=2nd dict
+        2nd dict:
+            keys=offset ranges, values=separated DataFrames
+    log_e_reco_bins: array-like. Returned if return_bins=True
+    offset_bins: array-like. Returned if return_bins=True
     """
 
     dtf_e_offset = dict()
 
     if offset_bins is None:
-        offset_bins = [0, max(dtf["camera_offset"])]  # Fit all elements in one bin by default.
-    elif any(o_bin < 0 for o_bin in offset_bins):
-        raise ValueError("Offset bins can't be negative.")
-    elif offset_bins[0] != 0:
-        raise ValueError("The first value of offset_bins must be zero.")
+        # n_offset_bins -1 from 0 to 4 deg.
+        offset_bins = np.linspace(0, 4, n_offset_bins)
+        # The last bin covers the rest of the camera-offset range. Up to 11 deg.
+        offset_bins = np.append(offset_bins, 11)
 
     if log_e_reco_bins is None:
         log_e_reco_bins = mstats.mquantiles(
-            dtf["log_reco_energy"].values, np.linspace(0, 1, n_bins)
+            dtf["log_reco_energy"].values, np.linspace(0, 1, n_e_bins)
```

Comment:
Doing this here means that the energy bins would have equal statistics before we bin if offset. Once we do, the energy bins might be quite uneven. It think it's fine and exactly what we want, but I just want to make sure it was a conscience choice and it's what we discussed in our call with @TarekHC.

Reply:
The binning in energy with equal statistics is only done for the training (before splitting the train and test samples). In this case, the offset binning is harmless, because it's only one big bin, so the statistics remain equal.
For the event-type partitioning though, the energy bins and the offset bins are both fixed and not made for having equal statistics. There are, in fact, huge differences between bins and we have to check that there are no empty bins (line 717).
I think this is intended, at least for the energy, because the energy binning is made to match the binning usually used in sensitivity curves. The offset binning so far is arbitrary.

Reply:
I'm afraid we will need to test what happens if we go for un-even equal-statistics bin in energy, for each uniform bin in offset. I feel this would be more realistic when creating the ET thresholds... But I also remember in the past we did this and we were getting weird stuff...

Maybe create an issue to test both approaches?

Reply:
I understand @JBernete's comment and completely agree with it that it doesn't matter since in training we anyway have just one large offset bin and later we use the standard 5-bin-per-decade binning. 

What I don't really understand is @TarekHC's comment. What do you want to test exactly? I assume the training is fine and you want to test different binnings for IRF production?